jdbc/src/main/java/org/apache/iotdb/jdbc/Utils.java

```diff
@@ -51,31 +51,17 @@ static IoTDBConnectionParams parseUrl(String url, Properties info)
       return params;
     }
     boolean isUrlLegal = false;
-    Pattern pattern = Pattern.compile("^"
-        + "(((?!-)[A-Za-z0-9-]{1,63}(?<!-)\\.)+[A-Za-z]{2,6}" // Domain name
-        + "|"
-        + "localhost" // localhost
-        + "|"
-        + "(([0-9]{1,3}\\.){3})[0-9]{1,3})" // Ip
-        + ":"
-        + "[0-9]{1,5}" // Port
-        + "/?$");
+    Pattern pattern = Pattern.compile("([^:]+):([0-9]{1,5})/?");
     String subURL = url.substring(Config.IOTDB_URL_PREFIX.length());
     Matcher matcher = pattern.matcher(subURL);
     if(matcher.matches()) {
       isUrlLegal = true;
     }
-    String[] DomainAndPort;
-    if(subURL.contains("/")) {
-      DomainAndPort = subURL.substring(0, subURL.length()-1).split(":");
-    } else {
-      DomainAndPort = subURL.split(":");
-    }
-    params.setHost(DomainAndPort[0]);
-    params.setPort(Integer.parseInt(DomainAndPort[1]));
     if (!isUrlLegal) {
-      throw new IoTDBURLException("Error url format, url should be jdbc:iotdb://domain|ip:port/ or jdbc:iotdb://domain|ip:port");
+      throw new IoTDBURLException("Error url format, url should be jdbc:iotdb://anything:port/ or jdbc:iotdb://anything:port");
     }
+    params.setHost(matcher.group(1));
```

Comment:
have you tried if the url is not matched?

Reply:
Add one test of `jdbc:iotdb//test6667`.